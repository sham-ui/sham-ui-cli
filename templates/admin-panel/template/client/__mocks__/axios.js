const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
