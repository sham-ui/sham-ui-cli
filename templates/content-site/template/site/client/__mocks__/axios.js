const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
