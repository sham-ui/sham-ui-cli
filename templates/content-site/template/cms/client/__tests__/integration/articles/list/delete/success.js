import setup, { app } from '../../../helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
} );

it( 'success delete article', async() => {
    expect.assertions( 4 );

    const DI = setup();
    axios
        .useDefaultMocks()
        .use( 'get', '/categories', {
            'categories': [
                { 'id': '1', 'name': 'Кухня', 'slug': 'kukhnia' },
                { 'id': '2', 'name': 'Быт', 'slug': 'byt' }
            ]
        } )
        .use( 'get', '/articles', {
            'articles': [
                {
                    'id': '10',
                    'title': 'Тест2',
                    'slug': 'test2',
                    'category_id': '1',
                    'published_at': '2022-04-25T19:34:23.619+07:00'
                },
                {
                    'id': '2',
                    'title': 'Вторая',
                    'slug': 'second',
                    'category_id': '2',
                    'published_at': '2022-03-22T08:26:23.619+07:00'
                }
            ],
            'meta': {
                'limit': 50, 'offset': 0, 'total': 2
            }
        } );

    history.pushState( {}, '', 'http://client.example.com/articles' );
    await app.start( DI );

    app.click( '[data-test-delete-button="10"]' );
    await app.waitRendering();

    app.checkMainPanel();

    axios
        .use( 'delete', '/articles/10' )
        .use( 'get', '/articles', {
            'articles': [
                {
                    'id': '2',
                    'title': 'Вторая',
                    'slug': 'second',
                    'category_id': '2',
                    'published_at': '2022-03-22T08:26:23.619+07:00'
                }
            ],
            'meta': {
                'limit': 50, 'offset': 0, 'total': 1
            }
        } );
    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();

    expect( axios.mocks.delete ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.delete.mock.calls[ 0 ][ 0 ] ).toBe( '/articles/10' );
    app.checkMainPanel();
} );
