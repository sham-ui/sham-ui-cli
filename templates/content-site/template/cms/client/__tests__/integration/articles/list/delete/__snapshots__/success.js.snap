// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success delete article 1`] = `
"<h4 class=\\"page-header\\"> Articles
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"articles-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container mg-b-30\\">
      <a data-test-toggle-create-form=\\"\\" class=\\"btn\\" href=\\"/articles/new\\"> Create new article </a>
    </div>
    <div class=\\"container\\">
      <div class=\\"article-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--0-->
            <table class=\\"list-table\\">
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Title</th>
                  <th>Slug</th>
                  <th>Category</th>
                  <th>Published at</th>
                  <th></th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td class=\\"text-center\\">10</td>
                  <td>Тест2</td>
                  <td>test2</td>
                  <td>Кухня</td>
                  <td>понедельник, 25 апреля 2022 г., 19:34:23</td>
                  <td class=\\"actions\\">
                    <a class=\\"btn\\" href=\\"/articles/10/edit\\" data-test-update-button=\\"10\\"> Edit </a>
                    <button class=\\"btn-danger\\" data-test-delete-button=\\"10\\"> Delete </button>
                  </td>
                </tr>
                <tr>
                  <td class=\\"text-center\\">2</td>
                  <td>Вторая</td>
                  <td>second</td>
                  <td>Быт</td>
                  <td>вторник, 22 марта 2022 г., 08:26:23</td>
                  <td class=\\"actions\\">
                    <a class=\\"btn\\" href=\\"/articles/2/edit\\" data-test-update-button=\\"2\\"> Edit </a>
                    <button class=\\"btn-danger\\" data-test-delete-button=\\"2\\"> Delete </button>
                  </td>
                </tr>
                <!--0-->
              </tbody>
            </table>
            <div class=\\"modal\\" data-test-modal=\\"\\"><span class=\\"close\\" title=\\"Close Modal\\" data-test-close-button=\\"\\">×</span>
              <!--0-->
              <div class=\\"modal-content\\">
                <div class=\\"container\\">
                  <h4>Delete article</h4>
                  <p class=\\"content\\">Are you sure you want to delete article \\"Тест2\\"?</p>
                  <div class=\\"footer\\">
                    <button data-test-ok-button=\\"\\" class=\\"btn-danger\\">Ok</button>
                    <button class=\\"btn-secondary\\" data-test-cancel-button=\\"\\">Cancel</button>
                  </div>
                </div>
              </div>
            </div>
            <!--0-->
            <!--1-->
            <!--0-->
            <!--0-->
            <!--1-->
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--2-->
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->"
`;

exports[`success delete article 2`] = `
"<h4 class=\\"page-header\\"> Articles
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"articles-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container mg-b-30\\">
      <a data-test-toggle-create-form=\\"\\" class=\\"btn\\" href=\\"/articles/new\\"> Create new article </a>
    </div>
    <div class=\\"container\\">
      <div class=\\"article-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--0-->
            <table class=\\"list-table\\">
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Title</th>
                  <th>Slug</th>
                  <th>Category</th>
                  <th>Published at</th>
                  <th></th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td class=\\"text-center\\">2</td>
                  <td>Вторая</td>
                  <td>second</td>
                  <td>Быт</td>
                  <td>вторник, 22 марта 2022 г., 08:26:23</td>
                  <td class=\\"actions\\">
                    <a class=\\"btn\\" href=\\"/articles/2/edit\\" data-test-update-button=\\"2\\"> Edit </a>
                    <button class=\\"btn-danger\\" data-test-delete-button=\\"2\\"> Delete </button>
                  </td>
                </tr>
                <!--0-->
              </tbody>
            </table>
            <!--1-->
            <!--0-->
            <!--0-->
            <!--1-->
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--2-->
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->"
`;
